application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/service/security/ApiKeyCatalogIdValidator.java

```diff
@@ -0,0 +1,33 @@
+package com.entando.hub.catalog.service.security;
+
+import com.entando.hub.catalog.persistence.entity.Catalog;
+import com.entando.hub.catalog.service.CatalogService;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApiKeyCatalogIdValidator {
+
+    private CatalogService catalogService;
+
+    public ApiKeyCatalogIdValidator(CatalogService catalogService) {
+        this.catalogService = catalogService;
+    }
+
+    public boolean validateApiKeyCatalogId(String apiKey, Long catalogId) {
+        if (StringUtils.isEmpty(apiKey) && null != catalogId) {
+            return false;
+        } else if (StringUtils.isNotEmpty(apiKey)) {
+            if (null == catalogId) {
+                return false;
+            }
```

Comment:
Consider unifying this logic in a single statement, something like
`if ((StringUtils.isEmpty(apiKey) && null != catalogId) || (StringUtils.isNotEmpty(apiKey) && null == catalogId))`
Could It be better to extract this logic to a meaningful method like `areBothApiKeyAndCatalogIdPresent()`?
Something like

```
if (! areBothApiKeyAndCatalogIdPresent(apiKey, catalogId)) {
  return false;
}
if (noApiKeyAndNoCatalogIdPresent(apiKey, catalogId)) {
  return true;
}

return [catalog checks]

```